### ISSUE TITLE
[False Negative] Analysis is skipped for closures causing missed bugs.

### ISSUE THREAD

**Issue by lynae99** [lizhuohua/rust-mir-checker#25]
`MirChecker` appears to completely skip the analysis of code blocks inside closures. This leads to false negatives for various types of bugs, ranging from memory safety violations (like Use-After-Free) to runtime panics (like Index-Out-Of-Bounds), if they happen to be wrapped in a closure. This also prevents the analyzer from detecting bugs in called functions that depend on specific arguments passed from within the closure.
**Reproduction**
```
unsafe fn deallocate<T>(ptr: *mut T, capacity: usize) {
    // The unsafe operation is wrapped in a closure
    (|| {
        let _vec: Vec<T> = Vec::from_raw_parts(ptr, 0, capacity);
    })()
}

fn bug(v: &mut Vec<u32>) {
    unsafe {
        deallocate(v.as_mut_ptr(), v.len());
    }
}

fn main() {
    let mut v = vec![1, 2, 3, 4, 5];
    bug(&mut v);
}
```
```
fn main() {
    (|| {
        let mut a = vec![1, 2, 3, 4, 5];
        let mut i = 0;
        while i < 5 {
            a[i] = i;
            i = i + 1;
        }
        let result = a[i]; 
    })()
}
```
```
fn main() {
    (|| {
    let _a = overflow(-1);
})()
}

fn overflow(time: i64) -> u32 {
    (time % 1000) as u32 * 1000000
}
```
**Expected Behavior** The tool should analyze the closure body and report:
1. A memory safety warning for Case 1.
2. An index-out-of-bounds warning for Case 2.
3. A provable overflow warning for Case 3.

**Actual Behavior**
In the above cases, `MirChecker` produces no warnings regarding the bugs inside the closures.
